lib/improver/spotdata/extract_data.py

```diff
@@ -319,56 +454,120 @@ def model_level_temperature_lapse_rate(self, cube, sites, neighbours,
             Cube of surface pressures at an equivalent time to the cube of
             screen level temperatures.
 
-        temperature_on_height_levels : iris.cube.Cube
-            Temperature data at 50m above the surface at an equivalent time to
-            the cube of screen level temperatures.
+        lower/upper_pressure : iris.cube.Cube
+            Cubes of pressure data at the defined lower and upper model
+            levels, each at an equivalent time to the cube of screen level
+            temperatures.
+
+        lower/upper_temperature : iris.cube.Cube
+            Cubes of temperature data at the defined lower and upper model
+            levels, each at an equivalent time to the cube of screen level
+            temperatures.
+
+        dz_tolerance : float
+            Vertical displacement between spotdata site and neighbouring grid
+            point below which there is no need to perform a lapse rate
+            adjustment.
+
+        dthetadz_threshold : float
+            Potential temperature gradient threshold, with gradients above this
+            value deemed to have been calculated across an inversion.
+
+        dz_max_adjustment : float
+            Maximum vertical distance over which a temperature will be adjusted
+            using the lapse rate. If the spotdata site is more than this
+            distance above or below its neighbouring grid point, the adjustment
+            will be made using dz = dz_max_adjustment.
 
         Returns:
         --------
+
         Data extracted from the screen level temperature cube at spotdata sites
         which has then been adjusted using a temperature lapse rate calculated
         using multi-level data.
 
         """
+        # Reference pressure of 1000hPa (1.0E5 Pa).
+        p_ref = 1.0E5
+
+        def _adjust_temperature(theta_base, dthetadz, dz, p_site, kappa):
+            """
+            Perform calculation of temperature at SpotData site.
+
+            """
+            if dz < 0 or dthetadz > 0:
+                # Case for extrapolating downwards or a stable atmosphere.
+                return (theta_base + dz*dthetadz)*(p_site/p_ref)**kappa
```

Comment:
The `theta_base + dz*dthetadz` are effectively calculating a `theta_site` variable. It might be slightly simpler to write:
```python
theta_site = theta_base + dz*dthetadz
return theta_site*(p_site/p_ref)**kappa
```
This would help make this equation more similar to the equation in the docstring.